sdk/helper/keysutil/policy.go

```diff
@@ -2393,3 +2397,165 @@ func wrapTargetPKCS8ForImport(wrappingKey *rsa.PublicKey, preppedTargetKey []byt
 	wrappedKeys := append(ephKeyWrapped, targetKeyWrapped...)
 	return base64.StdEncoding.EncodeToString(wrappedKeys), nil
 }
+
+func (p *Policy) CreateCsr(keyVersion int, csrTemplate *x509.CertificateRequest) ([]byte, error) {
+	keyEntry, err := p.safeGetKeyEntry(keyVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	if keyEntry.IsPrivateKeyMissing() {
+		// NOTE: Shouldn't this be a client error?
+		return nil, errutil.UserError{Err: "private key not imported for key version selected"}
+	}
+
+	// Template cannot be used as it is because it carries "signatureAlgorithm"
+	// information that might not apply to the key we are going to use.
+	// NOTE: The x509.CertificateRequest type in Go's crypto/x509 package does
+	// not directly include a field for storing the email address. However, the
+	// email address can be included as part of the subject distinguished name
+	// (DN) field in the certificate request.
+	csrTemplate = &x509.CertificateRequest{
+		Subject:         csrTemplate.Subject,
+		DNSNames:        csrTemplate.DNSNames,
+		EmailAddresses:  csrTemplate.EmailAddresses,
+		URIs:            csrTemplate.URIs,
+		Extensions:      csrTemplate.Extensions,
+		ExtraExtensions: csrTemplate.ExtraExtensions,
+		Attributes:      csrTemplate.Attributes, // Deprecated
+	}
```

Comment:
```suggestion
	csrTemplate.Signature = nil
	csrTemplate.SignatureAlgorithm = x509.UnknownSignatureAlgorithm
```

I think this should work for what we're doing. :-)